### PR TITLE
Task 5 "Authors list"

### PR DESCRIPTION
**Pull request by OlyaOlga**

### CODE REVIEW

**Comment by alexchmykhalo** [comment]
@OlyaOlga Please, merge "development" branch into your branch to get all latest changes. Seems there will be merge conflicts. Please, contntact me if you have any dificulties with that.

**Comment by OlyaOlga** [comment]
It seems now everything is ok with branch  feature/5-authors-list. There isn't any merge conflict no more.

File: src/AUTHORS.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<big><b><centre> The Kuzya  </centre> </b></big>
```
Comment (alexchmykhalo) [inline src/AUTHORS.txt]: Would you like to keep it as html? Then it is better to change extension of this file. But it also an option to use just simple text (and not to use rich text). What do you think.

If we will choose approach with simple text file - it is better to place it in the root of our repo, please see example here: https://github.com/django/django/blob/master/AUTHORS

File: src/kuzya.cpp [hunk 2]
```diff
@@ -1027,28 +1034,44 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
-                                                                                         "\n\n <b>Splashscreen Design:</b>      \n \t <centre>Oksana Rondyak</centre> "
-                                                                                         "\n <u>relax777@users.sourceforge.net</u> </pre>").arg(KUZYA_VERSION),QMessageBox::Ok,this,Qt::Dialog);
-    aboutBox->setIconPixmap(QPixmap(":/common/Kuzya_about.png"));
-    aboutBox->exec();
-    delete aboutBox;
+    const QString fileName = ":/AUTHORS.txt";
+    QFile file (fileName);
+    //checks file
+    {
+     if( ! QFile::exists(fileName))
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: Please, fix indentation.

File: src/kuzya.cpp [hunk 3]
```diff
@@ -1027,28 +1034,44 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
-                                                                                         "\n              \n \t <centre>Victor Sklyar</centre> "
-                                                                                         "\n <u>bouyantgrambler@users.sourceforge.net</u>"
-                                                                                         "\n              \n \t <centre>Alex Chmykhalo</centre> "
-                                                                                         "\n <u>alexchmykhalo@users.sourceforge.net</u>"
-                                                                                         "\n\n <b>Splashscreen Design:</b>      \n \t <centre>Oksana Rondyak</centre> "
-                                                                                         "\n <u>relax777@users.sourceforge.net</u> </pre>").arg(KUZYA_VERSION),QMessageBox::Ok,this,Qt::Dialog);
-    aboutBox->setIconPixmap(QPixmap(":/common/Kuzya_about.png"));
-    aboutBox->exec();
-    delete aboutBox;
+    const QString fileName = ":/AUTHORS.txt";
```
Comment (alexchmykhalo) [inline src/kuzya.cpp:1037]: Please, use curly braces for initialization.

File: src/kuzya.cpp [hunk 4]
```diff
@@ -1027,28 +1034,44 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
+    QFile file (fileName);
+    //checks file
+    {
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: Why you need this additional block here?

File: src/kuzya.cpp [hunk 5]
```diff
@@ -1027,28 +1034,44 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
+        }
+         if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+         {
+            qCritical()<<"couldn't open file"<<fileName;
+         }
+    }
+       QString textInf;
+       QDialog* information = new QDialog;
+       QVBoxLayout* layout = new QVBoxLayout(information);
+      information->setLayout(layout);
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: Please, fix indentation.

**Comment by OlyaOlga** [comment]
In AUTHORS file deleted all html, used simple text, moved file to the root of the repo. Took into account other corrections.

File: src/kuzya.cpp [hunk 6]
```diff
@@ -37,6 +37,12 @@
 #include <QSplitter>
 #include <QFileDialog>
 #include <QMessageBox>
+# include <QScrollArea>
```
Comment (alexchmykhalo) [inline src/kuzya.cpp:40]: Please, remove spaces after #

File: src/kuzya.cpp [hunk 7]
```diff
@@ -1027,28 +1034,39 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
-                                                                                         "\n <u>alexchmykhalo@users.sourceforge.net</u>"
-                                                                                         "\n\n <b>Splashscreen Design:</b>      \n \t <centre>Oksana Rondyak</centre> "
-                                                                                         "\n <u>relax777@users.sourceforge.net</u> </pre>").arg(KUZYA_VERSION),QMessageBox::Ok,this,Qt::Dialog);
-    aboutBox->setIconPixmap(QPixmap(":/common/Kuzya_about.png"));
-    aboutBox->exec();
-    delete aboutBox;
+    const QString fileName = ":/AUTHORS.txt";
+    QFile file (fileName);
+    //checks file
+    if( ! QFile::exists(fileName))
```
Comment (alexchmykhalo) [inline src/kuzya.cpp:1040]: Please, remove redundant spaces before and after "!"

**Review by alexchmykhalo — APPROVED** [review]
Please, fix style issues according to comments and merge.

**Pull request merged**